src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -244,6 +244,28 @@ void EIRPlantLoopHeatPump::doPhysics(EnergyPlusData &state, Real64 currentLoad)
     // evaluate capacity modifier curve and determine load side heat transfer
     Real64 capacityModifierFuncTemp =
         CurveManager::CurveValue(state, this->capFuncTempCurveIndex, loadSideOutletSetpointTemp, this->sourceSideInletTemp);
+
+    if (capacityModifierFuncTemp < 0.0) {
+        if (this->capModFTErrorIndex == 0) {
+            ShowSevereMessage(state, format("{} \"{}\":", DataPlant::PlantEquipTypeNames[static_cast<int>(this->EIRHPType)], this->name));
```

Comment:
If this pattern gets propagated any further, we should probably refactor it into a worker function that can be reused by many component models.